IndoorMapsBackend/src/BuildingResolver.ts

```diff
@@ -59,45 +102,110 @@ export class BuildingResolver {
             });
         }
-        return dbFloors.floors.map((value: DbFloor) => {
-            return {
-                ...value,
-                id: "floor" + value.id.toString()
-            }
-        })
+        return dbFloors.floors.map((value: DbFloor) => convertToGraphQLFloor(value))
     }
 
     @Mutation((returns) => Building)
     async createBuilding(
         @Arg('data') data: BuildingCreateInput,
         @Ctx() ctx: Context,
     ): Promise<Building> {
-        // TODO: only get the floors and areas if needed
-        // Use prisma to get building
+        const user = await validateUser(ctx.cookies);
+        if (!user) {
+            throw new GraphQLError('User not signed in', {
+                extensions: {
+                    code: 'BAD_USER_INPUT',
+                },
+            });
+        }
         const newBuilding = await ctx.prisma.building.create({
             data: {
                 title: data.title,
-                description: data.description,
+                address: data.address,
+                startLat: data.startLat,
+                startLon: data.startLon,
                 editors: {
                     create: {
                         editorLevel: "owner",
                         user: {
                             connect: {
-                                id: data.owner,
+                                id: user.databaseId,
                             },
                         },
                     },
                 },
+            }
+        });
+        return convertToGraphQLBuilding(newBuilding);
+    }
+
+    @Mutation((returns) => NewFloorResult)
+    async createFloor(
+        @Arg('data') data: FloorCreateInput,
+        @Ctx() ctx: Context,
+    ): Promise<NewFloorResult> {
+        const user = await validateUser(ctx.cookies);
+        if (!user) {
+            throw new GraphQLError('User not signed in', {
+                extensions: {
+                    code: 'BAD_USER_INPUT',
+                },
+            });
+        }
+        const newFloor = await ctx.prisma.floor.create({
+            data: {
+                title: data.title,
+                description: data.description,
+                building: {
+                    connect: {
+                        id: data.buildingDatabseId,
+                    },
+                },
+            },
+        });
+
+        return {
+            success: true,
+            databaseId: newFloor.id,
+            buildingDatabaseId: data.buildingDatabseId
+        };
+    }
+
+    @Mutation((returns) => NewFloorResult)
+    async modifyFloor(
+        @Arg('data') data: FloorModifyInput,
+        @Ctx() ctx: Context,
+    ): Promise<NewFloorResult> {
+        const user = await validateUser(ctx.cookies);
+        if (!user) {
+            throw new GraphQLError('User not signed in', {
+                extensions: {
+                    code: 'BAD_USER_INPUT',
+                },
+            });
+        }
```

Comment:
Since this code is repeated 3 times in this file, you can move the error throwing part inside validateUser?

Reply:
Great idea,
Wrapped getting a user or throwing an error in a function:
https://github.com/Theo-s-Meta-Capstone/IndoorMaps/commit/f733bda49e5faa823f5517488b67a7ad4d21dd13
Tested with apollo sandbox
<img width="1371" alt="image" src="https://github.com/Theo-s-Meta-Capstone/IndoorMaps/assets/26506274/d80f9819-2979-4531-ab93-364bd7f35964">